### PR TITLE
Improves local development experience

### PR DESCRIPTION
**Pull request by mchurichi**
This PR improves the local development experience by leveraging Docker to have a full-featured website without installing additional dependencies.

### CODE REVIEW

**Comment by mchurichi** [comment]
Hey @ajessup, thanks for taking a look at this!
The error you get is because you should have run `make docker-build` before trying to serve the website. Anyways, I just changed this and whenever you `make docker-serve` it will build the image as a previous step so you don't have to do it manually. Please take a look again now.

File: README.md [hunk 1]
```diff
@@ -11,32 +11,51 @@ The site is built using:
...
 
-To develop the locally, you'll first need to install some assets (mostly Sass assets) using [npm](https://npmjs.org):
+### Using Docker
 
+This is the preferred way to run the website for local development. 
 ```shell
-make setup
+$ make docker-serve
 ```
+This command will create a Docker image with all the tools you need to run the website. Note that some of the steps might ask you to run additional commands, please ignore them.
```
Comment (sanderson042) [inline README.md:20]:
```suggestion
This command takes a few minutes to create a Docker image with all the tools you need to run the website. Note that some of the steps might ask you to run additional commands, please ignore them. The following line indicates that the image is ready:
\`\`\`
Web Server is available at //localhost:1313/ (bind address 0.0.0.0)
\`\`\`
Open `http://localhost:1313`in your browser to view your local version of the spiffe.io website.
```

File: README.md [hunk 2]
```diff
@@ -11,32 +11,51 @@ The site is built using:
...
 
-With those assets in place, you can run the site locally using [Hugo](#hugo) or [Docker](#docker).
-
-### Hugo
-
-To run the site using Hugo, make sure it's [installed](https://gohugo.io/getting-started/installing/) and then run:
+Once the image is created, it will run the website on development mode while listening to changes in any of the markdown files or the [external content descriptor file](./external.yaml), rebuilding the site and reloading the website in your browser every time it detects changes.
 
+You can also run the website with the list of SPIRE releases on the Downloads page:
```
Comment (sanderson042) [inline README.md:24]:
```suggestion
You can also run the website with a list of releases under the "SPIRE Releases" heading on the Downloads page (`content/downloads/_index.md`):
```

**Review by sanderson042 — COMMENTED** [review]
Hi Max - here's a few updates that might help people who are new to the project.

**Comment by ajessup** [comment]
Thanks @mchurichi - I can confirm this is works for me now.

**Review by ajessup — APPROVED** [review]
Approved, pending @sanderson042's suggestions

**Comment by mchurichi** [comment]
Thanks for the review @sanderson042!, all of your suggestions have been applied, some of them with a few minor changes, please take a look one more time.

**Review by sanderson042 — APPROVED** [review]
Looks great! Thank you, Max 🚀

**Pull request merged**